### PR TITLE
feat: more context menu actions

### LINKED ISSUE #1123: I can't delete views in enterprise version
**Describe the bug**
I can't delete views, I added some views and now I want delete them and I can't, when I left click on top on the view name nothing happen

**To Reproduce**

1. Go to new view created on enterprise version
2. Click left on the name on top
3. Nothing happen we can't delete the view

**Expected behavior**
Possibility to delete the new view

**Screenshots**
CleanShot 2024-12-01 at 15 46 03

** Client (please complete the following information):**
- OS: macos
- Browser chrome
- Version 131.0.6778.86 

**Platform (Please tell us which deployment version you are using)**
image: ghcr.io/teableio/teable-ee

Other info
I don't have any authority role

### PR DESCRIPTION
**Pull request by kpodp0ra**
Closes #1123

1. Added context menu to space selection: 

2. Added right-click action to open this context menu
3. Added right-click action to table selection menu

5. Replaced onBlur with useClickAway, because table rename via context menu caused troubles in Firefox
6. Added right-click action to views
obraz
7. Made PopoverTrigger slightly bigger there, so you can click on view's button edge and still open context menu

### CODE REVIEW

**Comment by kpodp0ra** [comment]
Closes #1123

File: apps/nextjs-app/src/features/app/blocks/space/space-side-bar/SpaceItem.tsx [hunk 1]
```diff
@@ -14,26 +19,85 @@ interface IProps {
...
+      queryClient.invalidateQueries({ queryKey: ReactQueryKeys.space(space.id) });
+    },
+  });
+
+  useEffect(() => {
+    if (isEditing) setTimeout(() => inputRef.current?.focus());
+  }, [isEditing]);
+
+  useClickAway(inputRef, async () => {
+    if (isEditing && inputRef.current?.value && inputRef.current.value !== space.name)
```
Comment (caoxing9) [inline apps/nextjs-app/src/features/app/blocks/space/space-side-bar/SpaceItem.tsx]: better to enclose the if statement
```suggestion
    if (isEditing && inputRef.current?.value && inputRef.current.value !== space.name) {
    await updateSpaceMutator({
        spaceId: space.id,
        updateSpaceRo: { name: inputRef.current.value },
      });
}
```

File: apps/nextjs-app/src/features/app/blocks/table-list/TableListItem.tsx [hunk 2]
```diff
@@ -43,6 +46,12 @@ export const TableListItem: React.FC<IProps> = ({ table, isActive, className, is
     }
   }, [isEditing]);
 
+  useClickAway(inputRef, () => {
+    if (isEditing && inputRef.current?.value && inputRef.current.value !== table.name)
```
Comment (caoxing9) [inline apps/nextjs-app/src/features/app/blocks/table-list/TableListItem.tsx]: the same with above

File: apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx [hunk 3]
```diff
@@ -135,19 +129,25 @@ export const ViewListItem: React.FC<IProps> = ({ view, removable, isActive }) =>
...
-          {commonPart}
+          {isActive && showViewMenu ? (
+            <PopoverTrigger asChild>{commonPart}</PopoverTrigger>
+          ) : (
+            <PopoverAnchor asChild>{commonPart}</PopoverAnchor>
+          )}
         </Button>
         <PopoverContent className="w-32 p-1">
-          <div className="flex flex-col">
+          {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-static-element-interactions */}
```
Comment (caoxing9) [inline apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx]: Do not ignore the rule, using role = 'button' etc attributes setting you can search in project. It improve user experience.

Comment (kpodp0ra) [reply apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx]: @caoxing9 you sure? This div shouldn't appear as clickable. 

You have the same case in an existing codebase, and you don't mark it as a button either:

https://github.com/teableio/teable/blob/a5898a5be4dbfac8dd6e350864fb3e8d5345d764/apps/nextjs-app/src/features/app/blocks/table-list/TableListItem.tsx#L62-L63

this code is a workaround for JS's strange mechanics, not something that gives the user some functionality

Comment (caoxing9) [reply apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx]: You're right, i misunderstood.

File: apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx [hunk 4]
```diff
@@ -135,19 +129,25 @@ export const ViewListItem: React.FC<IProps> = ({ view, removable, isActive }) =>
         }
         navigateHandler();
       }}
+      onContextMenu={() => showViewMenu && setOpen(true)}
```
Comment (Sky-FE) [inline apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx:134]: Here we need to consider the interaction in touch devices.

Comment (kpodp0ra) [reply apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx:134]: @Sky-FE Right-click is just a new, alternative way to open this menu. The old method of double-clicking haven't got deleted and is compatible with mobile devices.

Comment (Sky-FE) [reply apps/nextjs-app/src/features/app/blocks/view/list/ViewListItem.tsx:134]: Oh, i misunderstood.

**Review by Sky-FE — APPROVED** [review]

**Comment by caoxing9** [comment]
<video src="https://github.com/user-attachments/assets/50efd0b8-d527-42e8-94ed-99cd66d0a48e" controls>
</video>
A little flaw, when click rename, popover content flash in the upper left corner.
Eesy to reproduct.

**Comment by kpodp0ra** [comment]
Interesting :eyes:. I will try to fix it on my free time.

**Comment by kpodp0ra** [comment]
Rebased PR onto develop branch. Ready to merge.

**Review by caoxing9 — APPROVED** [review]

**Pull request merged**